app/src/main/java/com/selfdrivingboat/boatcontroller/SelfDrivingBoatCamera.java

```diff
@@ -116,9 +129,12 @@ public void onImageAvailable(ImageReader reader) {
 
         private void save(byte[] bytes) throws IOException {
             OutputStream output = null;
+            String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US).format(new Date());
             try {
                 output = new FileOutputStream(file);
                 output.write(bytes);
+                StorageReference reference = FirebaseStorage.getInstance().getReference().child(timeStamp);
+                UploadTask uploadTask = reference.putBytes(bytes);
```

Comment:
nice this looks good to me! do you understand what bytes is? is basically the image in form of 0101010 if you print it you will get some HEX code (AE EF E9 E2) that is just a concise way to express binary.